reconcile/utils/saasherder/saasherder.py

```diff
@@ -1615,25 +1619,36 @@ def get_container_images_diff_saas_file(
         for rt in saas_file.resource_templates:
             for target in rt.targets:
                 try:
-                    if not target.image:
+                    if not (target.image or target.images):
                         continue
+                    if target.image and target.images:
+                        raise TriggerSpecContainerImageError(
+                            '"image" and "images" are mutually exclusive. Do not set both in the same saas target.'
+                        )
                     commit_sha = self._get_commit_sha(
                         url=rt.url,
                         ref=target.ref,
                         github=github,
                     )
-                    desired_image_tag = commit_sha[: rt.hash_length or self.hash_length]
-                    # don't trigger if image doesn't exist
-                    image_registry = f"{target.image.org.instance.url}/{target.image.org.name}/{target.image.name}"
-                    image_uri = f"{image_registry}:{desired_image_tag}"
                     image_auth = self._initiate_image_auth(saas_file)
+                    desired_image_tag = commit_sha[: rt.hash_length or self.hash_length]
+
+                    all_images = [target.image] if target.image else target.images or []
```

Comment:
since already verified `image` and `images` can only have one set, can simplify `if else or` with

```python
    all_images = target.images or [target.image]
```